### PR TITLE
[Feature:Forum] Inline attachments

### PR DESCRIPTION
**Pull request by andrewaikens87**
### What is the current behavior?
Attachments are displayed as popups.

### What is the new behavior?
Attachments are now displayed inline for the forum. They are still clickable to bring up the full popup. Currently the width is limited to 400px.
all toggle
view

### CODE REVIEW

**Comment by bmcutler** [comment]
@andrewaikens87 Please fix conflicts & failing tests

File: site/public/js/forum.js [hunk 1]
```diff
@@ -1054,4 +1054,66 @@ function sortTable(sort_element_index, reverse=false){
...
\ No newline at end of file
+}
+
+function formatIFrameInlineImages(a, name, url){
+    //Not best practice but good enough for V1
+    return function(){
+        var contents = $('#file_viewer_' + a + '_iframe').contents();
+        var body = contents.find('body');
+        var img = body.find('img');
+        body.append(`<style>
```
[the remark this answers is not in the document]
Comment (andrewaikens87) [reply site/public/js/forum.js]: Discussed in weekly meeting.

**Comment by xprilion** [comment]
@andrewaikens87 I was trying to test run this, and it seems it displays very long images as very, very long images! Maybe a max-height too would be required..

File: site/public/js/forum.js [hunk 2]
```diff
@@ -1156,6 +1155,69 @@ function loadThreadHandler(){
...
+          img:hover { box-shadow: 0 0 2px 1px rgba(0, 140, 186, 0.5); cursor: pointer; } </style>`);
+      body.css("text-align", "center");
+      body.append('<h4 style="text-align:center">' + decodeURI(name) + '</h3> <br/>');
+      iframe.height(contents.find('img').height() + 100);
+      $(img).attr("title", "Click to view attachment in a popup");
+      $(img).click(function() {
+        window.open(url,"_blank","toolbar=no,scrollbars=yes,resizable=yes, width=700, height=600");
+      });
+    }
+}
```
Comment (sak6lab) [inline site/public/js/forum.js]: What is the best practice for this?

**Comment by sak6lab** [comment]
Is it beneficial to instead make another function in MiscController that returns the image that is called by an ajax call when the button is clicked. The ajax callback would then add an image tag to the attachment well?

**Comment by sak6lab** [comment]
Didn't mean to close it

**Comment by andrewaikens87** [comment]
:thumbsup:

**Pull request merged**